lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2015,13 +2015,17 @@ int prArrayLace(struct VMGlobals *g, int numArgsPushed)
 	numLists = obj1->size;
 
 	if(IsNil(b)) {
+		// If length argument is unspecified, compute length as the multiple of
+		// the length of the shortest sub-array and the number of sub-arrays.
 		for (j=0; j<numLists; ++j) {
 			slot = slots + j;
 			if(isKindOfSlot(slot, class_array))	{
 				len = slotRawObject(slot)->size;
 				if(j==0 || n>len) { n = len; }
 			} else {
-				return errFailed; // this primitive only handles Arrays.
+				// If not an array we assume a max per-element length of one.
+				n = 1;
+				break;
```

Comment:
Should the for loop really terminate at the first instance of a non-array? Don't we try to find the maximum length of all?

Reply:
Since we're trying to find the minimum length (see above) we can safely terminate.

Reply:
ok, so `[1, [1, 2], [1, 2, 3]].lace` would return `[1]`? It would have to be `[1, 1, 1]` according to the current behavior. `[[1], [1, 2], [1, 2, 3]].lace` returns `[1, 1, 1]`.

Reply:
```[1, [1, 2], [1, 2, 3]].lace``` would before this CL return an error, and now returns
```[1, 1, 1]``` like you would expect. Remember that the length is the product of the minimum length and the number of elements in the array, so in that case it would be ```1 * 3 = 3```.